sandbox/apparmor/notify/strings.go

```diff
@@ -78,3 +131,28 @@ func (su *stringUnpacker) UnpackString(offset uint32) (string, error) {
 	}
 	return string(tmp[:idx]), nil
 }
+
+// UnpackStrings unpacks N contiguous NUL-terminated strings at a given offset
+// into the buffer.
+func (su *stringUnpacker) UnpackStrings(offset uint32, n uint32) ([]string, error) {
+	if offset == 0 {
+		return nil, nil
+	}
+	if offset >= uint32(len(su.Bytes)) {
+		return nil, fmt.Errorf("address %d points outside of message body", offset)
+	}
+	if n == 0 {
+		return nil, nil
+	}
+	strs := make([]string, n)
+	for i := uint32(0); i < n; i++ {
+		tmp := su.Bytes[offset:]
+		idx := bytes.IndexByte(tmp, 0)
+		if idx < 0 {
+			return nil, fmt.Errorf("unterminated string at address %d", offset)
+		}
+		strs[i] = string(tmp[:idx])
+		offset += uint32(idx + 1) // advance offset to start of next string
```

Comment:
I think this is more precise, no?
```suggestion
		offset += uint32(idx) + 1 // advance offset to start of next string
```